src/Visualizations/TimeSeries.js

```diff
@@ -0,0 +1,314 @@
+        // The attributes 'date' and 'scrobbles' are used to get the content
+        // of the tooltip when hovering on a cell
+        .attr("date", formatDate(dates[`${i}`]))
+        .attr("scrobbles", 0)
+        .on("mouseover", mouseover)
+        .on("mousemove", mousemove)
+        .on("mouseleave", mouseleave);
+    }
+  }
+
+  /**
+   * React function to run when the component updates
+   *
+   * It's expected that the component will be updated every time UserPage
+   * fetches one more page of the scrobbles list.
+   */
+  componentDidUpdate() {}
+
+  /**
+   * React function to run when the component has been mounted
+   *
+   * This runs only once, and uses that fact to initialize the canvas, and append
+   * the items that won't be redrawn.
+   */
+  componentDidMount() {
+    // This creates the svg for the graph
+    const svg = d3
+      .select(`#d3-section-${this.props.title}`)
+      .append("svg")
+      // This makes the svg element responsive
+      // @see https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/preserveAspectRatio
+      // @see https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/viewBox
+      .attr("preserveAspectRatio", "xMinYMin meet")
+      .attr("viewBox", "-20 -119 640 120");
+    // .attr("viewBox", "-20 -250 560 270");
+
+    // // Draw the Heatmap structure: Legend, texts, and cells
+    // this.drawLegend(svg);
+    // this.drawWeekDays(svg);
+    this.drawHeatmapCells(svg);
+
+    // Initialize the tooltip. Draw it after all other items so it stays at top
+    tooltip = svg.append("rect");
+    tooltip_text = svg.append("text");
+
+    // Assign the props value to a variable to avoid writing a lot
+    let lastList = this.props.user.scrobbles;
+
+    // Convert the list of scrobble objects into a list of strings using the
+    // date to determine an ID
+    let idList = [].concat.apply(
+      [],
+      lastList.map((e) =>
+        e.list
+          .map((e) =>
+            getIDFromDate(
+              new Date(
+                1000 * Math.ceil(Number(e.date.uts) / 604800) * 604800 -
+                  259200000
+              )
+            )
+          )
+          .filter((e) => this.frequencyList[`${e}`] !== undefined)
+      )
+    );
+
+    // This counts the number of times that a given id appears in the
+    // current list of ids
+    for (let i = 0; i < idList.length; i++) {
+      let id = idList[`${i}`];
+      // If this.frequencyList.id is defined, increment its value
+      if (this.frequencyList[`${id}`] !== undefined) {
+        this.frequencyList[`${id}`]++;
+      }
+    }
+
+    // Update the values considering the new added scrobbles
+    this.updateHeatmapValues();
+  }
+
+  render() {
+    return (
+      <>
+        <div
+          id={`d3-section-${this.props.title}`}
+          className={styles["graph-container"]}
+        />
+      </>
+    );
+  }
+}
+
+/* ---------------------------- Helping Functions --------------------------- */
```

Comment:
As these are shared with heatmap, they should be in a separate file `../utils/{script}.js`